src/charging-station/Bootstrap.ts

```diff
@@ -154,13 +156,37 @@ export default class Bootstrap {
         poolOptions: {
           workerChoiceStrategy: Configuration.getWorkerPoolStrategy(),
         },
-        messageHandler: async (msg: ChargingStationWorkerMessage) => {
-          if (msg.id === ChargingStationWorkerMessageEvents.STARTED) {
-            this.uiServer.chargingStations.add(msg.data.id as string);
-          } else if (msg.id === ChargingStationWorkerMessageEvents.STOPPED) {
-            this.uiServer.chargingStations.delete(msg.data.id as string);
-          } else if (msg.id === ChargingStationWorkerMessageEvents.PERFORMANCE_STATISTICS) {
-            await this.storage.storePerformanceStatistics(msg.data as unknown as Statistics);
+        messageHandler: (msg: InternalChargingStationWorkerMessage) => {
+          const workerEventStarted = (data: ChargingStationData) => {
+            this.uiServer.chargingStations.set(data.hashId, data.data);
+            ++this.numberOfChargingStations;
+          };
+          const workerEventStopped = (data: ChargingStationData) => {
+            this.uiServer.chargingStations.delete(data.hashId);
+            --this.numberOfChargingStations;
+          };
+          const workerEventPerformanceStatistics = (data: Statistics) => {
+            (async () => this.storage.storePerformanceStatistics(data))().catch((error: unknown) =>
```

Comment:
No need for async if you chain the promise to a .catch(), async/await is just syntactic sugar for 
`myPromise.then().catch()`

Reply:
It doesn't work, because the function return: `void | Promise<void>`.
Since it can return `void` typescript won't let me try catch it.

Reply:
It looks more like a type definition issue.

Reply:
It is, but if i change the type by removing `void` it collides with the definition of `storePerformanceStatistics` in `JsonFileStorage`